src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -156,7 +156,7 @@
     /**
      * Set of building modules this building has.
      */
-    protected Map<Class<? extends IBuildingModule>, IBuildingModule> modules = new LinkedHashMap<>();
+    protected List<IBuildingModule> modules = new ArrayList<>();
```

Comment:
I had it like this already, but then it failed in a few of the use cases I remember. Can you double check if the existing modules still work fine after this change?